netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -138,6 +140,11 @@ public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
 
       SSLEngine sslEngine = sslContext.newEngine(ctx.alloc());
       ctx.pipeline().addFirst(new SslHandler(sslEngine, false));
+
+      // TODO(lukaszx0) Seems like there might be a better way to pass this object other than
+      // setting attribute on channel?
+      Attribute<SSLSession> sslSessionAttr = ctx.channel().attr(Utils.SSL_SESSION_ATTR_KEY);
```

Comment:
This logic wants to live below, after we've recieved the event that handshake has actually completed. Otherwise, we'll get a bogus session. See https://docs.oracle.com/javase/7/docs/api/javax/net/ssl/SSLEngine.html#getSession()